spec/factories/user_roles.rb

```diff
@@ -2,15 +2,15 @@
 
 FactoryBot.define do
   factory :user_role do
-    user { FactoryBot.create(:user) }
+    user { rand(2) == 0 ? FactoryBot.create(:user_with_wca_id) : FactoryBot.create(:user) }
```

Comment:
I don't like randomness in our test suite. Why do you flip a coin here?

Reply:
Thing is, I want to make sure this works irrespective of whether the user has a WCA ID or not. Do you have any alternative suggestion?

Reply:
two tests?

Reply:
I was thinking to test both randomly in each and every test where user_role is created. Just like how we create random name and random date. Adding 2 tests for each tests can be painful, so I've decided not to test the user_with_wca_id case. I've removed that.

Reply:
There is a distinct difference between randomness in test _data_, and randomness in test _cases_.
For example, when you create a dummy competition for tests, the latitude and longitude will be random. That's okay because you're only using this competitions as "some valid competition that I don't really care about", and once you have created it, the randomized lat/long stays the same. It's generated randomly once, saved into the mocked object, and then it stays like that. (If you run into a test case where you care about the exact lat/long, like testing distances in the Competition Series proximity, you will see that these tests very carefully compute distinct lat/long values instead of relying on randomness).
Your random call actually influences test _behavior_, i.e. the thing that you're testing is wildly different. That's not good and it's also the reason why I flagged it during review.

Reply:
Oh okay, understood. Thanks a lot for the detailed explanation. :-)